src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -418,7 +419,44 @@ func (c controller) postInstallConfigs(ctx context.Context) error {
 		if err = c.updatePendingOLMOperators(); err != nil {
 			return errors.Errorf("Timeout while waiting for some of the operators and not able to update its state")
 		}
-		c.log.Warnf("Timeout while waiting for OLM operators be installed")
+		c.log.WithError(err).Warnf("Timeout while waiting for OLM operators be installed")
+		// Don't return the error here so the installation is marked as sucessful.
+		// We only log the error, and update the state of the monitored operators in assisted-service API.
+	}
+
+	// Aplly post install manifests:
+	err = c.applyPostInstallManifests()
+	if err != nil {
+		c.log.WithError(err).Warnf("Failed to apply post manifests.")
+	}
+
+	return nil
+}
+
+func (c controller) applyPostInstallManifests() error {
+	ctx := utils.GenerateRequestContext()
+	dest, err := ioutil.TempFile("", "custom_manifests")
+	defer dest.Close()
+	defer os.Remove(dest.Name())
+	if err != nil {
+		return err
+	}
+
+	if err := c.ic.DownloadFile(ctx, "custom_manifests.yaml", dest.Name()); err != nil {
+		return err
+	}
+
+	// download kubeconfig file
+	kubeconfigFileName := "kubeconfig-noingress"
```

Comment:
don't we have such code already? if we do lets have it in one place. I any case lets move it to function

Reply:
Yeah, there is must-gather, I will make it reusable.